src/calculations.hpp

```diff
@@ -55,4 +55,22 @@ bool is_in_direction_of_target(
  */
 double distSquared(const tf2::Transform & a, const tf2::Transform & b);
 
+/**
+ * @brief Closest point between a line segment and a point
+ *
+ * Calculate the closest point between the line segment bounded by `segment_start` and `segment_end`
+ * and point `point`.
+ *
+ * @param[in] point               The point
+ * @param[in] segment_start       Start of the line segment
+ * @param[in] segment_end         End of the line segment
+ * @param[in] estimate_pose_angle Indicates if the pose angle should be estimated from the line
+ *                                segment (true) or if the pose angle from segment_start should be
```

Comment:
I just realized that probably a better idea is to interpolate the angle from start to end (when those angles are given). This will avoid "step" discontinuities of the returned angle in this function between segments (which I saw recently in one of our projects). I will create an issue for this